scripts/generate_pipeline_files.py

```diff
@@ -0,0 +1,169 @@
+                    "Select Resources Needed: BigQuery (BQ), Google Cloud Storage (GCS)?"
+                ),
+                type=click.Choice(["BQ", "GCS"], case_sensitive=False),
+                default="r",
+            )
+            if resource == "BQ":
+                bq_yaml = sample_yaml["resources"][0]
+                bq_yaml["dataset_id"] = dataset_id
+                bq_desc = click.prompt(
+                    "A user-friendly description of the dataset", type=str
+                )
+                bq_yaml["description"] = bq_desc
+                resources.append(bq_yaml)
+            if resource == "GCS":
+                gcs_yaml = sample_yaml["resources"][1]
+                gcs_bucket_name = click.prompt(
+                    "Your Cloud Storage Bucket Name\n"
+                    "Use hyphenated syntax, e.g. `some-prefix-123`, for the names.\n"
+                    "Note that bucket names must not contain 'google' or close misspellings, such as 'g00gle'.",
+                    type=str,
+                )
+                location = click.prompt(
+                    (
+                        "The location of the bucket.\n"
+                        "Object data for objects in the bucket resides in physical storage within this region.\n"
+                        "Defaults to US."
+                    ),
+                    type=click.Choice(["US", "EU", "ASIA"], case_sensitive=False),
+                    default="US",
+                )
+                gcs_yaml["name"] = gcs_bucket_name
+                gcs_yaml["location"] = location
+                resources.append(gcs_yaml)
+            if not click.confirm("Would you like to add another resource"):
+                output["resources"] = resources
+                with open(f"{dir}/dataset.yaml", "w") as dataset_out:
+                    yaml.dump(output, dataset_out, Dumper=yaml.Dumper, sort_keys=False)
+                break
+
+
+def create_pipeline_yaml(dir: str, pipeline_id: str):
+    sample_yaml = None
+    output = {}
+    resources = []
+    with open(f"{PROJECT_ROOT}/samples/pipeline.yaml", "r") as sample_pipeline_yaml:
+        try:
+            sample_yaml = yaml.safe_load(sample_pipeline_yaml)
+            tables = click.prompt(
+                "Input your BigQuery Table name(s) required for your pipeline\n"
+                "If you have multiple tables, please use a comma-seperated list. (eg. table1, table2, table3)"
+            )
+            for table_name in tables.split(","):
+                print(table_name)
+                sample_yaml["resources"][0]["table_id"] = table_name
+                bq_resource = sample_yaml["resources"][0]
+                resources.append(bq_resource.copy())
+            output["resources"] = resources
+
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    tasks = []
+    with open(f"{PROJECT_ROOT}/scripts/dag_imports.json", "r") as dag_imports:
```

Comment:
Let's use pathlib here so the rest of the function isn't indented:
```
dag_imports = (PROJECT_ROOT / "scripts" / "dag_imports.json").read_text()

# ...
```

Reply:
updated here: https://github.com/GoogleCloudPlatform/public-datasets-pipelines/pull/412/commits/9080e14727d66ba4533b6d3dfb8bf577b2ce5939#diff-b760de9bd0fdeea47e69e074deb04cb2ef5604a0ae70ef0fa58b04121d206318R139-R141